runtime/onert/core/src/odc/CodegenManager.cc

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "CodegenLoader.h"
+#include "odc/CodegenManager.h"
+#include "util/Utils.h"
+
+#include <fstream>
+#include <mutex>
+
+namespace onert
+{
+namespace odc
+{
+
+bool CodegenManager::codegen(const char *target, CodegenPreference pref)
```

Comment:
I'm not sure what `target` is but code that is checking the `target` is necessary, I think.

Reply:
`target` is the term in compiler. `source` and `target`. You may think it as the binary of codegen. I agree to add check.

Reply:
You can find the detail explanation of `target` string in header file.

```
   * @param target  Target backend name
   *                This target string will be used to find a backend library.
   *                The name of target backend library should follow the following rules:
   *                  'lib' + {backend extension} + '-gen' + {lib extension}
   *                And the target string should be a name except 'lib' and {lib extension}.
   *                For example, if the backend extension is 'aaa', the backend library name
   *                should be 'libaaa-gen.so', and the target string should be 'aaa-gen'.
```

This Codegen manager dynamically loads the target library. To find a shared library, a search policy must be defined. In this case, the library is searched for using this target string.